FluxNotes/flux
# Support completing attribute shortcut labels via Context Portal

_Pull requests into Flux require the following. Submitter should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:. Reviewers should complete every item on the bulleted list when reviewing._

Can now complete attribute labels (e.g. #status). Currently, the functionality works similar to something like @condition. You can type #status in the editor and then the picker portal will appear to select a value. I did this by adding a picker option to the shortcuts in `Shortcuts.json` as well as update the triggers we manually set in `ShortcutManager` to use said picker.

Things not covered that will be covered by other tasks:

* Completion of attributes that support multiple options (e.g. #reasons)
* Ability to close the context portal and re-open for selection when your cursor is next to an incomplete attribute.

Test case:

1) type #disease status (should show up as incomplete)
2) type #status (#disease status should now be complete, #status should be incomplete and context portal should open)
3) Select a value, should replace #status with the selected value.


## Submitter:

**Running the Application**


**Task**


**Documentation**


**Code Quality**


**Tests**



## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code
